src/Altinn.Platform/Altinn.Platform.Authorization/IntegrationTests/PolicyAdministrationPointTest.cs

```diff
@@ -157,6 +157,64 @@ public async Task TryDeleteDelegationPolicyRules_Valid()
+        {
+            // Arrange
+            int performedByUserId = 20001336;
+            int offeredByPartyId = 50001337;
+            int coveredBy = 20001337;
+            string coveredByType = AltinnXacmlConstants.MatchAttributeIdentifiers.UserAttribute;
+            _delegationMetadataRepositoryMock.MetadataChanges = new Dictionary<string, List<DelegationChange>>();
+
+            List<RequestToDelete> inputRuleMatchess = new List<RequestToDelete>
+            {
+                TestDataHelper.GetRequestToDeleteModel(performedByUserId, offeredByPartyId, "org1", "app5", new List<string> { "0d0c8570-64fb-49f9-9f7d-45c057fddf94", "6f11dd0b-5e5d-4bd1-85f0-9796300dfded" }, coveredByUserId: coveredBy),
+                TestDataHelper.GetRequestToDeleteModel(performedByUserId, offeredByPartyId, "org2", "app3", new List<string> { "244278c1-7c6b-4f6b-b6e9-2bd41f84812f" }, coveredByUserId: coveredBy),
+                TestDataHelper.GetRequestToDeleteModel(performedByUserId, offeredByPartyId, "org1", "app4", new List<string> { "adfa64fa-5859-46e5-8d0d-62762082f3b9" }, coveredByUserId: coveredBy)
+            };
+
+            List<Rule> expected = new List<Rule>
+            {
+                TestDataHelper.GetRuleModel(performedByUserId, offeredByPartyId, coveredBy.ToString(), coveredByType, "Read", "org2", "app3", createdSuccessfully: true),
+                TestDataHelper.GetRuleModel(performedByUserId, offeredByPartyId, coveredBy.ToString(), coveredByType, "Read", "org1", "app4", createdSuccessfully: true)
+            };
+
+            Dictionary<string, List<DelegationChange>> expectedDbUpdates = new Dictionary<string, List<DelegationChange>>
+            {
+                { "org2/app3/50001337/u20001337", new List<DelegationChange> { TestDataHelper.GetDelegationChange("org2/app3", offeredByPartyId, performedByUserId, isDeleted: false, coveredByUserId: coveredBy) } },
+                { "org1/app4/50001337/u20001337", new List<DelegationChange> { TestDataHelper.GetDelegationChange("org1/app4", offeredByPartyId, performedByUserId, isDeleted: false, coveredByUserId: coveredBy) } }
+            };
+
+            // Act
+            List<Rule> actual = await _pap.TryDeleteDelegationPolicyRules(inputRuleMatchess);
+
+            // Assert
+            Assert.Equal(expected.Count, actual.Count);
+            Assert.True(actual.All(r => r.CreatedSuccessfully));
+            Assert.True(actual.All(r => !string.IsNullOrEmpty(r.RuleId)));
+            AssertionUtil.AssertEqual(expected, actual);
+            AssertionUtil.AssertEqual(expectedDbUpdates, _delegationMetadataRepositoryMock.MetadataChanges);
+            _logger.Verify(
+                x => x.Log(
+                    LogLevel.Warning,
+                    It.IsAny<EventId>(),
+                    It.Is<It.IsAnyType>((@object, @type) => @object.ToString() == "The policy is already deleted for App: org1/app5 CoveredBy: 20001337 OfferedBy: 50001337"),
```

Comment:
Nice! Didn't expect to see a test verifying logging. It does mean you can easily test how it works. I'm pretty sure the `LogWarning` method ignores token names and only use the order. In other words, what we have between the `{}` characters in the string is irrelevant. This might allow you to simplify `{policyToDelete.PolicyMatch.OfferedByPartyId}` to `{offeredByPartyId}`.

Reply:
I Ignore this